kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/discovery/provider/KubernetesServiceInstanceServiceProvider.java

```diff
@@ -99,7 +99,8 @@ public String getMode() {
                         LOG.error("Error while processing discovered service [" + serviceName + "]", throwable);
                     }
                     return new ArrayList<>();
-                });
+                })
+                .defaultIfEmpty(new ArrayList<>());
```

Comment:
Same behaviour of the service instance providers. There can still be situation when the Service is returned however due to configuration the result can be empty.